hydrolib/core/io/structure/models.py

```diff
@@ -214,11 +228,13 @@ def validate(cls, v):
         return super().validate(v)
 
     def _exclude_fields(self) -> Set:
-        # exclude the unset props like coordinates or branches
-        if self.branchid is not None:
-            exclude_set = {"numcoordinates", "xcoordinates", "ycoordinates"}
+        # exclude the non-applicable, or unset props like coordinates or branches
+        if self.type is "compound":
+            exclude_set = self._loc_all_fields
```

Comment:
More readable to use `_loc_coord_fields | _loc_branch_fields` here?

Reply:
I prefer to keep it like this: it reflects that a compound never has ay coordinate fields. And it is also consistent with the check_location code on line 106.